tools/mbedtls.cmake

```diff
@@ -0,0 +1,85 @@
+include(FetchContent)
+
+set(MBEDTLS_2_VERSION 2.28.0)
+
+FetchContent_Declare(
+    mbedtls_2
+    GIT_REPOSITORY "https://github.com/ARMmbed/mbedtls"
+    GIT_TAG v${MBEDTLS_2_VERSION}
+    PATCH_COMMAND ${MODULE_ROOT_DIR}/tools/mbedtls_configure.sh <SOURCE_DIR> config.h
+)
+
+if(NOT mbedtls_2_POPULATED)
+    FetchContent_Populate(mbedtls_2)
+
+    set(MBEDTLS_2_BIN_DIR ${CMAKE_CURRENT_BINARY_DIR}/lib/mbedtls_2)
+    set(MBEDTLS_TARGET_PREFIX "MbedTLS2_")
+
+    option(USE_STATIC_MBEDTLS_LIBRARY "" ON)
+    option(USE_SHARED_MBEDTLS_LIBRARY "" OFF)
+    option(ENABLE_PROGRAMS "" OFF)
+    option(ENABLE_TESTING "" OFF)
+
+    add_subdirectory(${mbedtls_2_SOURCE_DIR} ${mbedtls_2_BINARY_DIR})
+
+    add_library(MbedTLS2_interface INTERFACE)
+    get_target_property(mbedtls_includes MbedTLS2_mbedtls INCLUDE_DIRECTORIES)
+    target_include_directories(
+        MbedTLS2_interface
+        INTERFACE ${mbedtls_includes}
+        INTERFACE ${mbedtls_2_SOURCE_DIR}/library
+        INTERFACE ${mbedtls_2_SOURCE_DIR}/include/mbedtls
+    )
+
+    set_target_properties(
+        MbedTLS2_mbedcrypto MbedTLS2_mbedtls MbedTLS2_mbedx509
+        PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${MBEDTLS_2_BIN_DIR} LIBRARY_OUTPUT_DIRECTORY
+                                                                 ${MBEDTLS_2_BIN_DIR}
+    )
+
+    add_library(MbedTLS2::mbedtls ALIAS MbedTLS2_mbedtls)
```

Comment:
Do these libraries not need to be set if mbedtls is already populated? Or is the mbedtls_2_POPULATED check above not for checking if it was already downloaded?

Reply:
Looks like i'm missing a call to FetchContent_GetProperties here. Will add.

Reply:
mbedtls_2_POPULATED is not a cache variable, and is set by FetchContent_GetProperties and FetchContent_Populate.